client/src/components/Enter.tsx

```diff
@@ -4,7 +4,11 @@ import { useParams, useNavigate } from "react-router-dom";
 import Navbar from "../partials/NavBar";
 import Footer from "../partials/Footer";
 
-const Enter = ({ signIn }) => {
+interface EnterProps {
```

Comment:
Use `type` in most places where you don't need to extend

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces